distributed/tests/test_worker.py

```diff
@@ -2337,7 +2337,7 @@ def raise_exc(*args):
         h.key: "memory",
         res.key: "error",
     }
-    await assert_task_states_on_worker(expected_states_B, b)
+    await assert_task_states_on_worker(expected_states_B, b), b.tasks
```

Comment:
```suggestion
    await assert_task_states_on_worker(expected_states_B, b)
```
see my comment above